src/librustc_typeck/check/regionck.rs

```diff
@@ -1795,7 +1795,7 @@ impl<'a, 'gcx, 'tcx> RegionCtxt<'a, 'gcx, 'tcx> {
                     let (outlives, _) =
                         self.replace_late_bound_regions_with_fresh_var(
                             span,
-                            infer::AssocTypeProjection(projection_ty.item_name(self.tcx)),
+                            infer::AssocTypeProjection(self.tcx.associated_item(projection_ty.item_def_id).name),
```

Comment:
SGTM, but I think we can keep pulling loose ends forever and it's already a nightmare to keep track of comments in this PR. I'd be more comfortable wrapping it up at this point and happy to pick up whatever clean ups you envision individually. Easier that way.